pennylane/_qubit_device.py

```diff
@@ -791,6 +801,42 @@ def mutual_info(self, wires0, wires1, log_base):
             state, indices0=wires0, indices1=wires1, c_dtype=self.C_DTYPE, base=log_base
         )
 
+    def classical_shadow(self, wires, n_snapshots, circuit, seed=None):
+        """TODO: docs"""
+        if circuit is None:  # pragma: no cover
+            raise ValueError("Circuit must be provided when measuring classical shadows")
+
+        with set_shots(self, shots=1):
+            # slow implementation but works for all devices
+            n_qubits = len(self.wires)
+            device_wires = np.array(self.map_wires(wires))
+
+            if seed is not None:
+                rng = np.random.RandomState(seed)
+                recipes = rng.randint(0, 3, size=(n_snapshots, n_qubits))
+            else:
+                recipes = np.random.randint(0, 3, size=(n_snapshots, n_qubits))
+            obs_list = [qml.PauliX, qml.PauliY, qml.PauliZ]
+
+            outcomes = np.zeros((n_snapshots, n_qubits))
+
+            for t in range(n_snapshots):
+                # compute rotations for the Pauli measurements
+                rotations = [
+                    rot
+                    for wire in wires
+                    for rot in obs_list[recipes[t][device_wires[wire]]].compute_diagonalizing_gates(
+                        wires=wire
+                    )
+                ]
+
+                self.reset()
+                self.apply(circuit.operations, rotations=circuit.diagonalizing_gates + rotations)
+
+                outcomes[t] = self.generate_samples()[0]
+
+        return self._cast(self._stack([outcomes, recipes]), dtype=np.uint8)
```

Comment:
Originally I had it to match the ADR, but I haven't tested this with the other interfaces yet. I still think it would be good to return ints here simply because it's more intuitive. (Both the bits and recipes are discrete valued.)

Reply:
I added tests for the other interfaces and for `default.mixed`. It looks like a bug in the `_as_array` method causing an unwanted cast.

Reply:
Marking this as resolved since all devices now return `int`.